src/components/DataTable/DataTable.vue

```diff
@@ -0,0 +1,319 @@
+<template>
+  <table class="data-table">
+    <thead class="data-table__head">
+      <tr class="data-table__header-mobile">
+        <th class="data-table__table-header-mobile">
+          <slot name="header-sort" :sortColumn="sortColumn">
+            <Button @click="sortColumn(internalSortByColumn)">Sort: A-Z</Button>
```

Comment:
I'm in favor of the slot exposing `sort` as a method rather than `sortColumn`. 

`internalSortBy` is the string value for the column.
`internalSortByColumn` is the full column object.

the sorting method `sortColumn` accepts a column as its argument.